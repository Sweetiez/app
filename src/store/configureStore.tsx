import {createStore, combineReducers} from 'redux';
import cartReducer from './reducers/cart';
import shopReducer from './reducers/shop';
import userReducer from './reducers/user';
import ordersReducer from './reducers/orders';
import recipesReducer from './reducers/recipes';

const rootReducer = combineReducers({
  cart: cartReducer,
  shop: shopReducer,
  user: userReducer,
  orders: ordersReducer,
  recipes: recipesReducer,
});
const confStore = () => {
  return createStore(rootReducer);
};
export default confStore;
