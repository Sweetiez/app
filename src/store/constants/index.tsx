// cart
export const ADD_ITEM_TO_CART = 'ADD_ITEM_TO_CART';
export const REMOVE_ITEM_FROM_CART = 'REMOVE_ITEM_FROM_CART';
export const CLEAR_CART = 'CLEAR_CART';

// shop
export const UPDATE_SHOP = 'UPDATE_SHOP';
export const GET_PRODUCTS = 'GET_PRODUCTS';
export const PRODUCT_ERROR = 'PRODUCT_ERROR';

// user
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const REGISTER = 'REGISTER';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const REGISTER_OK = 'REGISTER_OK';
export const REGISTER_ERROR = 'REGISTER_ERROR';
export const SET_USER = 'SET_USER';
export const GET_USER_ERROR = 'GET_USER_ERROR';
export const UPDATE_OK = 'UPDATE_OK';
export const UPDATE_ERROR = 'UPDATE_ERROR';
export const UPDATE_PASSWORD_OK = 'UPDATE_PASSWORD_OK';
export const UPDATE_PASSWORD_ERROR = 'UPDATE_PASSWORD_ERROR';

// evaluation
export const COMMENT_ERROR = 'COMMENT_ERROR';
export const COMMENT_OK = 'COMMENT_OK';
export const REPORT_OK = 'REPORT_OK';
export const REPORT_ERROR = 'REPORT_ERROR';
export const REPORT_ALREADY_MADE = 'REPORT_ALREADY_MADE';

// orders
export const GET_ORDERS_ERROR = 'GET_ORDERS_ERROR';
export const SET_ORDERS = 'SET_ORDERS';
export const CREATE_AN_ORDER_ERROR = 'CREATE_AN_ORDER_ERROR';
export const CREATE_PAYMENT_INTENT_ERROR = 'CREATE_AN_ORDER_ERROR';

// recipes
export const UPDATE_RECIPES = 'UPDATE_RECIPES';
export const RECIPES_ERROR = 'RECIPES_ERROR';
export const RECIPE_ERROR = 'RECIPE_ERROR';
