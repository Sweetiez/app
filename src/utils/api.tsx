import Config from 'react-native-config';
import {Platform} from 'react-native';

export const getUrl = (path: string) => {
  if (Platform.OS === 'ios') {
    return Config.API_IOS_LOCAL_URL + path;
  } else {
    return Config.API_URL + path;
  }
};

export const buildHeaders = token => {
  return {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: token ? 'Bearer ' + token : undefined,
  };
};

export const buildRequest = async (
  method,
  url,
  data,
  onError,
  onSuccess,
  token,
) => {
  const headers = buildHeaders(token);
  try {
    const response = await fetch(getUrl(url), {
      method: method,
      headers,
      body: JSON.stringify(data),
    });
    if (response.status === 200 || response.status === 201) {
      return onSuccess(response);
    } else {
      return onError();
    }
  } catch (error) {
    return onError();
  }
};
