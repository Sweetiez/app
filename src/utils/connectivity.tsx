import NetInfo from '@react-native-community/netinfo';

export const checkConnectivity = async () => {
  const status = await NetInfo.fetch();
  return status.isConnected;
};
