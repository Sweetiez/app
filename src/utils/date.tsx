//@ts-ignore
import moment from 'moment';

export const formatDate = (date: string) => {
  return moment(date).format('DD/MM/YYYY');
};
export const formatDashDate = (date: string) => {
  return moment(date).format('YYYY-MM-DD');
};
