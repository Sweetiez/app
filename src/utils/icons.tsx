// @ts-nocheck
import React from 'react';
import Pen from './../assets/icons/pen.svg';
import Minus from './../assets/icons/minus.svg';
import Plus from './../assets/icons/plus.svg';
import Add from './../assets/icons/add.svg';
import Back from './../assets/icons/arrow-back.svg';
import CookBook from './../assets/icons/cook-book.svg';
import Cake from './../assets/icons/cake.svg';
import Event from './../assets/icons/event.svg';
import Account from './../assets/icons/account.svg';
import Eye from './../assets/icons/eye.svg';
import Interrogation from './../assets/icons/interrogation.svg';
import Cart from './../assets/icons/cart.svg';
import Trash from './../assets/icons/trash.svg';
import NoImage from './../assets/icons/no-image.svg';
import Edit from './../assets/icons/edit.svg';
import Warning from './../assets/icons/warning.svg';
import Orders from './../assets/icons/orders.svg';
import Filter from './../assets/icons/filter.svg';
import Cross from './../assets/icons/cross.svg';
import Circle from './../assets/icons/circle.svg';
import colors from '../assets/colors';

const getIcons = (iconName, color = colors.white, size = 18) => {
  const props = {
    width: size,
    height: size,
    color: color,
  };
  switch (iconName) {
    case 'pen':
      return <Pen {...props} />;
    case 'eye':
      return <Eye {...props} />;
    case 'minus':
      return <Minus {...props} />;
    case 'plus':
      return <Plus {...props} />;
    case 'add':
      return <Add {...props} />;
    case 'back':
      return <Back {...props} />;
    case 'cake':
      return <Cake {...props} />;
    case 'event':
      return <Event {...props} />;
    case 'cookBook':
      return <CookBook {...props} />;
    case 'account':
      return <Account {...props} />;
    case 'interrogation':
      return <Interrogation {...props} />;
    case 'cart':
      return <Cart {...props} />;
    case 'trash':
      return <Trash {...props} />;
    case 'noImage':
      return <NoImage {...props} />;
    case 'edit':
      return <Edit {...props} />;
    case 'warning':
      return <Warning {...props} />;
    case 'orders':
      return <Orders {...props} />;
    case 'filter':
      return <Filter {...props} />;
    case 'cross':
      return <Cross {...props} />;
    case 'circle':
      return <Circle {...props} />;
  }
};

export default getIcons;
