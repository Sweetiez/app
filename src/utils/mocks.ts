jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');
jest.mock('./../assets/icons/arrow-back.svg', () => 'Back');
jest.mock('./../assets/icons/cook-book.svg', () => 'CookBook');
jest.mock('./../assets/icons/cake.svg', () => 'Cake');
jest.mock('./../assets/icons/event.svg', () => 'Event');
jest.mock('./../assets/icons/eye.svg', () => 'Eye');
jest.mock('./../assets/icons/trash.svg', () => 'Trash');
jest.mock('./../assets/icons/no-image.svg', () => 'NoImage');
jest.mock('./../assets/icons/edit.svg', () => 'Edit');
jest.mock('./../assets/icons/warning.svg', () => 'Warning');

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}))



jest.mock('react-native-picker-select', () => 'RNPickerSelect')
jest.mock('react-native-config', () => 'Config')