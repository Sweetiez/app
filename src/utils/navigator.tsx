import getIcons from './icons';

const getTabIcon = (route, focused, color, size, t) => {
  switch (route.name) {
    case t('tabBar.shop'):
      return getIcons('cake', color, 24);
    case t('tabBar.recipes'):
      return getIcons('cookBook', color, 24);
    case t('tabBar.events'):
      return getIcons('event', color, 24);
    case t('tabBar.account'):
      return getIcons('account', color, 24);
    case t('tabBar.cart'):
      return getIcons('cart', color, 24);
  }
};

export default getTabIcon;
