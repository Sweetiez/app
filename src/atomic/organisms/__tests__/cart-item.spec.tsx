import React from 'react';
import renderer from 'react-test-renderer';
import '../../../utils/mocks';
import ReduxProvider from '../../../utils/provider';
import CartItem from '../cart-item';
import {store} from '../../../store/configureStore';

describe('CartItem', () => {
  it('renders correctly', () => {
    const cartItem = {
      quantity: 5,
      item: {
        name: 'Tartelette',
        description: 'This is a description',
        images: ['https://picsum.photos/1220/300'],
        rating: 5,
        price: 10,
      },
    };
    const tree = renderer
      .create(
        <ReduxProvider reduxStore={store}>
          <CartItem cartItem={cartItem} />
        </ReduxProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
