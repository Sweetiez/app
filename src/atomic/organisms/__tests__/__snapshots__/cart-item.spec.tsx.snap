// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "shadowColor": "rgba(0, 0, 0, 0.07)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://picsum.photos/1220/300",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 15,
          "borderTopLeftRadius": 15,
          "height": 100,
          "marginRight": 10,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="#000000"
      size={22}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 22,
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textDecoration="none"
      textTransform="none"
    >
      Tartelette
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 10,
          },
        ]
      }
    />
    <Text>
      cart.quantity
      5
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 10,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      50
      common.euros
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 10,
          "zIndex": 1000,
        }
      }
    >
      <Trash
        color="#ff0000"
        height={20}
        width={20}
      />
    </View>
  </View>
</View>
`;
